pints/tests/test_toy_stochastic_degradation_model.py

```diff
@@ -0,0 +1,106 @@
+#
+import unittest
+import numpy as np
+import pints
+import pints.toy
+
+
+class TestStochasticDegradation(unittest.TestCase):
+    """
+    Tests if the stochastic degradation (toy) model works.
+    """
+
+    def test_start_with_zero(self):
+        # Test the special case where the initial concentration is zero
+        model = pints.toy.StochasticDegradationModel(0)
+        times = [0, 1, 2, 100, 1000]
+        parameters = [0.1]
+        values = model.simulate(parameters, times)
+        self.assertEqual(len(values), len(times))
+        self.assertTrue(np.all(values == np.zeros(5)))
+
+    def test_start_with_twenty(self):
+        # Run small simulation
+        model = pints.toy.StochasticDegradationModel(20)
+        times = [0, 1, 2, 100, 1000]
+        parameters = [0.1]
+        values = model.simulate(parameters, times)
+        self.assertEqual(len(values), len(times))
+        self.assertEqual(values[0], 20)
+        self.assertEqual(values[-1], 0)
+
+    def test_suggested(self):
+        model = pints.toy.StochasticDegradationModel(20)
+        times = model.suggested_times()
+        parameters = model.suggested_parameters()
+        self.assertTrue(np.all(times == np.linspace(0, 100, 101)))
+        self.assertEqual(parameters, 0.1)
+        values = model.simulate(parameters, times)
+        self.assertEqual(values[0], 20)
+        self.assertEqual(values[-1], 0)
+
+    def test_simulate(self):
+        model = pints.toy.StochasticDegradationModel(20)
+        parameters = [0.1]
+        times = np.linspace(0, 100, 101)
+        values = model.simulate(parameters, times)
+
+        # Test output of Gillespie algorithm
+        self.assertTrue(np.all(self._mol_count ==
+                               np.array(range(20, -1, -1))))
+
+        # Test interpolation function
+        # Check exact time points from stochastic simulation
+        self.assertTrue(np.all(self._interp_func(self._time) ==
+                               self._mol_count))
+
+        # Check simulate function returns expected values
+        self.assertTrue(np.all(values[np.where(times < self._time[1])] == 20))
+
+        # Check interpolation function works as expected
+        self.assertTrue(self._interp_func(np.random.uniform(self._time[0],
+                                          self._time[1])) == 20)
+        self.assertTrue(self._interp_func(np.random.uniform(self._time[1],
+                                          self._time[2])) == 19)
+
+    def test_errors(self):
+        model = pints.toy.StochasticDegradationModel(20)
+        # parameters, times cannot be negative
+        times = np.linspace(0, 100, 101)
+        parameters = -0.1
+        self.assertRaises(ValueError, model.simulate, parameters, times)
+        self.assertRaises(ValueError, model.deterministic_mean, parameters,
+                          times)
+        self.assertRaises(ValueError, model.deterministic_variance, parameters
+                          times)
```

Comment:
Syntax error here!!! `...parameters *,* times)`! Make sure you run all tests locally before you push it here... Then it will tell you the coverage (which we want it to be 100%).